runtime/lua/vim/_editor.lua

```diff
@@ -651,7 +651,7 @@ do
   end
 end
 
-local on_key_cbs = {} --- @type table<integer,function>
+local on_key_cbs = {} --- @type table<integer,[function, {}?]>
```

Comment:
This annotation may be inaccurate. Currently `vim.on_key()` only checks if `opts` is a table. It doesn't check the content of `opts`, so `opts` is not necessarily an empty table.

Reply:
I'd like to keep the _private implementation_ a list assuming there will eventually be `vim.on_key()` options; or at least until it's decided there will be no options.  The code that works with a list is not that complex, so taking it out and putting it back soon is not that big of a deal. But in any event probably
```
local on_key_cbs = {} --- @type table<integer,[function, table?]>
```
makes more sense for the private data structure.

I can take the `{opts}` out of the public facing
```
vim.on_key({fn}, {ns_id}, {opts})
```
But I'm hoping that soon there's a PR with `{mapping = boolean?}`, (default true). BTW, I believe that would be the last missing piece for a full (pretty full) popup implementation.

Any thoughts on adding `{mapping = boolean?}` and getting the information from `lua` to `C`.

Reply:
This is not big deal, but there is another thing: It may be better to use `opts = opts or {}` in `vim.on_key()`, so that `opts` is never `nil` in the table here.

Reply:
> Any thoughts on adding {mapping = boolean?} and getting the information from lua to C.

I think putting this in `vim.on_key()` may be a bit confusing, considering that disabling mappings is a global flag. But I have no idea where else to put it either.

Reply:
The vim doc only recommends using it in limited situations when all keys are consumed by the filter. If this guidance is used, there's no issue.
```
							*popup-mapping*
Normally the key is what results after any mapping, since the keys pass on as
normal input if the filter does not use it.  If the filter consumes all the
keys, set the "mapping" property to zero so that mappings do not get in the
way.  This is default for |popup_menu()| and |popup_dialog()|.
```

Reply:
> This is not big deal, but there is another thing: It may be better to use `opts = opts or {}` in `vim.on_key()` (like other Lua APIs do), so that `opts` is never `nil` in the table here, and then you can remove the `?`.

Cool. Good to know.

Reply:
> I think putting this in `vim.on_key()` may be a bit confusing, considering that disabling mappings is a global flag,

Yeah, the docs should say something like `typically only false when all keys are consumed`.

Reply:
DONE (small edits, nothing about mapping)